File: src/ELibrary.Backend/LibraryShopEntities/Domain/Entities/Shop/OrderBook.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LibraryShopEntities.Domain.Entities.Shop
{
    public class OrderBook
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public string Id { get; set; }
        [Required]
        [Range(1, int.MaxValue, ErrorMessage = "Book amount must be greater than 0")]
        public int BookAmount { get; set; }
        [Required]
        public int BookId { get; set; }
        [Required]
        public decimal BookPrice { get; set; }
        [Required]
        public int OrderId { get; set; }
        [ForeignKey("OrderId")]
        public Order Order { get; set; } = default!;
    }
}

File: src/ELibrary.Frontend/src/app/modules/shared/domain/models/shop/order/orderBook.ts
import { Book, CartBook, getDefaultBook } from "../../../..";

export interface OrderBook {
    id: string;
    bookAmount: number;
    bookPrice: number;
    bookId: number;
    book: Book;
}

export function getDefaultOrderBook(): OrderBook {
    return {
        id: "",
        bookAmount: 0,
        bookPrice: 0,
        bookId: 0,
        book: getDefaultBook()
    }
}
export function mapCartBookToOrderBook(cartBook: CartBook): OrderBook {
    return {
        id: cartBook.id,
        bookPrice: cartBook.book.price,
        bookAmount: cartBook.bookAmount,
        bookId: cartBook.bookId,
        book: cartBook.book
    };
}